### PR TITLE
Support for Airflow 2

### PR DESCRIPTION
**Pull request by agomez-etsy**
## Description

> Please include a brief summary of the changes.

## Context / Why are we making this change?

> Please include details on why this change is necessary, and any applicable tickets, design docs, or documentation links.

## Testing and QA Plan

> How has this work been tested or QA'd?

## Impact

> What are the implications of these changes? Are there any cross-cutting concerns to keep in mind?

### CODE REVIEW

File: boundary_layer/builders/base.py [hunk 1]
```diff
@@ -207,6 +213,23 @@ def render_target(self, node):
...
+        node_q.put(node)
+        downstream_operators = []
+
+        while not node_q.empty():
+            for downstr in self.graph.downstream_dependency_set(node_q.get()):
+                if downstr.requires_resources and downstr not in downstream_operators:
+                    upstr = [u for u in self.graph.upstream_dependency_set(downstr) if u != node]
+                    downstr.upstream_operators = upstr
+                    downstream_operators.append(downstr)
+                    node_q.put(downstr)
```
Comment (vchiapaikeo) [inline boundary_layer/builders/base.py:229]: You fancy

**Review by vchiapaikeo — APPROVED** [review]
This is awesome. What do you think about merging this branch to your forked repo for now? And even considering building a build process to your repo so we can operator directly on main from it?

File: boundary_layer/builders/templates/primary_preamble.j2 [hunk 2]
```diff
@@ -48,16 +47,19 @@ from {{ module }} import {{ ', '.join(objects) }}
 {% endif %}
 {% endfor %}
 
+from etsy.utils.defaults import patch_all
+patch_all()
```
Comment (eap) [inline boundary_layer/builders/templates/primary_preamble.j2:51]: What are these used for? Is this a private library?

Comment (eap) [reply boundary_layer/builders/templates/primary_preamble.j2:51]: @agomez-etsy - Friendly ping!

**Pull request closed without merging**